src/transforms/reduce/transform.rs

```diff
@@ -760,4 +782,100 @@ merge_strategies.bar = "concat"
         })
         .await;
     }
+
+    #[tokio::test]
+    async fn strategy_path_with_nested_fields() {
+        let reduce_config = toml::from_str::<ReduceConfig>(indoc!(
+            r#"
+            group_by = [ "id" ]
+
+            merge_strategies.id = "discard"
+            merge_strategies."message.a.b" = "array"
+
+            [ends_when]
+              type = "vrl"
+              source = "exists(.test_end)"
+            "#,
+        ))
+        .unwrap();
+
+        assert_transform_compliance(async move {
+            let (tx, rx) = mpsc::channel(1);
+
+            let (topology, mut out) = create_topology(ReceiverStream::new(rx), reduce_config).await;
+
+            let e_1 = LogEvent::from(Value::from(btreemap! {
+                "id" => 777,
+                "message" => btreemap! {
+                    "a" => btreemap! {
+                        "b" => [1,2],
+                        "num" => 1,
+                    }
+                },
+                "arr" => vec![btreemap! { "a" => 1 }, btreemap! { "b" => 1 }]
+            }));
+            let mut metadata_1 = e_1.metadata().clone();
+            metadata_1.set_upstream_id(Arc::new(OutputId::from("reduce")));
+
+            tx.send(e_1.into()).await.unwrap();
+
+            let e_2 = LogEvent::from(Value::from(btreemap! {
+                "id" => 777,
```

Comment:
Could we use a different value here to ensure that it is discarded?

Reply:
It's used as the group field here, so it will break the test(though definitely worth testing if it has not been covered yet).

Reply:
Ah I see 👍

Reply:
Indeed. The discard strategy is covered by the `"arr"` field in `strategy_path_with_nested_fields()`.

Reply:
I had the same confusion as @jszwedko though :) so might be worth removing this "discard" strategy, since it's not testing really anything.